### PR TITLE
[Fix-16567] Unable to delete copied nodes in unsaved workflow

### LINKED ISSUE #16567: [Bug] Unable to delete copied nodes in unsaved workflow
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

If a node is copied in an existing workflow, it cannot be deleted without being saved

image

### What you expected to happen

Copy a workflow node

### How to reproduce

1. Create a workflow with nodes
2. Copy the node without saving it
3. Delete a node (cannot be deleted)

### Anything else

_No response_

### Version

3.2.x

use  docker image: apache/dolphinscheduler-standalone-server:3.2.2

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Xuxiaotuan**
fix https://github.com/apache/dolphinscheduler/issues/16567
If a node is copied in an existing workflow, it cannot be deleted without being saved

## Brief change log
1. Fix the handling logic for when the backend API returns an empty array
2. If the dependent Lineage Task List is empty, no further query is required

## Pull Request Notice
[Pull Request Notice](https://github.com/apache/dolphinscheduler/blob/dev/docs/docs/en/contribute/join/pull-request.md)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Review by SbloodyS — COMMENTED** [review]
CI failed. Please fix it. @Xuxiaotuan

**Comment by Xuxiaotuan** [comment]
Read the wrong message is: " Error:  The archive file /home/runner/.m2/repository/com/github/eirslett/node/16.13.1/node-16.13.1-linux-x64.tar.gz is corrupted and will be deleted. Please try the build again."

CI may need to be executed again, and Quality Gate failure  I don't know what to do with this

File: dolphinscheduler-ui/src/views/projects/components/dependencies/use-dependencies.ts [hunk 1]
```diff
@@ -30,14 +30,21 @@ export function useDependencies() {
...
+      await queryDependentTasks(projectCode, dependentTaskReq).then(
+        (res: any) => {
+          if (res?.data?.length > 0) {
+            res.data
+              .filter(
+                (item: any) => item.processDefinitionCode !== workflowCode
+              )
+              .forEach((item: any) => {
+                tasks.push(
+                  item.processDefinitionName + '->' + item.taskDefinitionName
```
Comment (SbloodyS) [inline dolphinscheduler-ui/src/views/projects/components/dependencies/use-dependencies.ts]:
```suggestion
                (item: any) => item.workflowDefinitionCode !== workflowCode
              )
              .forEach((item: any) => {
                tasks.push(
                  item.workflowDefinitionName + '->' + item.taskDefinitionName
```

**Comment by SbloodyS** [comment]
> Read the wrong message is: " Error: The archive file /home/runner/.m2/repository/com/github/eirslett/node/16.13.1/node-16.13.1-linux-x64.tar.gz is corrupted and will be deleted. Please try the build again."
> 
> CI may need to be executed again, and Quality Gate failure I don't know what to do with this

Node related error are caused by network fluctuations, and you don't need to care about `Quality Gate` in this PR.

**Comment by Xuxiaotuan** [comment]
The front-end and code standards, along with the suggestions for modification, have been revised. Assistance is needed to execute the CI @SbloodyS

**Comment by SbloodyS** [comment]
> The front-end and code standards, along with the suggestions for modification, have been revised. Assistance is needed to execute the CI @SbloodyS

This still unaddressed. https://github.com/apache/dolphinscheduler/pull/16576#discussion_r1744678064

**Comment by Xuxiaotuan** [comment]
> > The front-end and code standards, along with the suggestions for modification, have been revised. Assistance is needed to execute the CI @SbloodyS
> 
> This still unaddressed. [#16576 (comment)](https://github.com/apache/dolphinscheduler/pull/16576#discussion_r1744678064)

I didn't notice, there was a PR that had already changed the field name, I have revised and resubmitted it.

**Review by SbloodyS — DISMISSED** [review]
This fix does not address the root cause, which is that the lineage info is not copyed when the workflow is copyed. Would you like to open another PR to fix it? @Xuxiaotuan 

This PR is LGTM. Thanks.

**Comment by SbloodyS** [comment]
Please run `pnpm run lint` to format the frontend code. @Xuxiaotuan

**Comment by Xuxiaotuan** [comment]
> 此修复未解决根本原因，即在复制工作流程时未复制沿袭信息。您是否要打开另一个PR来修复此问题？@Xuxiaotuan
> 
> 这个PR太棒了。谢谢。

Okay, I will open another PR to solve the issue（backend）.

**Comment by Xuxiaotuan** [comment]
> 此修复未解决根本原因，即在复制工作流程时未复制沿袭信息。您是否要打开另一个PR来修复此问题？@Xuxiaotuan
> 
> 我们将努力使您的梦想成真。谢谢。

Ok, I will open another PR to solve the root issue.

**Comment by Xuxiaotuan** [comment]
I don't know how to click on it  “dismissed”. I'm sorry. It wasn't my intention

**Comment by SbloodyS** [comment]
> I don't know how to click on it “dismissed”. I'm sorry. It wasn't my intention

It wasn't your clicking. Dissmiss will happen every time you repush the code. This is normal in the process of PR repair.

**Review by SbloodyS — APPROVED** [review]

**Review by ruanwenjun — APPROVED** [review]
+1

**Pull request merged**